xbmc/music/MusicDatabase.cpp

```diff
@@ -6232,204 +6487,331 @@ std::string CMusicDatabase::GetItemById(const std::string &itemType, int id)
+                  iFailCount++;
+                }
+              }
+            }
+            if (settings.m_artwork)
+            {
+              if (GetArtForItem(artist.idArtist, MediaTypeArtist, artwork))
+              {
+                std::string savedThumb = URIUtils::AddFileToFolder(strPath, "folder.jpg");
+                std::string savedFanart = URIUtils::AddFileToFolder(strPath, "fanart.jpg");
+                if (artwork.find("thumb") != artwork.end() && (settings.m_overwrite || !CFile::Exists(savedThumb)))
+                  CTextureCache::GetInstance().Export(artwork["thumb"], savedThumb);
+                if (artwork.find("fanart") != artwork.end() && (settings.m_overwrite || !CFile::Exists(savedFanart)))
+                  CTextureCache::GetInstance().Export(artwork["fanart"], savedFanart);
+              }
+            }
+            xmlDoc.Clear();
+            TiXmlDeclaration decl("1.0", "UTF-8", "yes");
+            xmlDoc.InsertEndChild(decl);
           }
-          xmlDoc.Clear();
-          TiXmlDeclaration decl("1.0", "UTF-8", "yes");
-          xmlDoc.InsertEndChild(decl);
         }
-      }
-
-      if ((current % 50) == 0 && progress)
-      {
-        progress->SetLine(1, CVariant{artist.strArtist});
-        progress->SetPercentage(current * 100 / total);
-        progress->Progress();
-        if (progress->IsCanceled())
+        if ((current % 50) == 0 && progressDialog != NULL)
         {
-          progress->Close();
-          m_pDS->close();
-          return;
+          progressDialog->SetLine(1, CVariant{ artist.strArtist });
+          progressDialog->SetPercentage(current * 100 / total);
         }
+        current++;
       }
-      current++;
     }
 
-    xmlDoc.SaveFile(xmlFile);
-
-    CVariant data;
-    if (singleFile)
+    if (settings.GetSinglefile())
     {
-      data["file"] = xmlFile;
-      if (iFailCount > 0)
-        data["failcount"] = iFailCount;
+      std::string xmlFile = URIUtils::AddFileToFolder(strFolder, "kodi_musicdb" + CDateTime::GetCurrentDateTime().GetAsDBDate() + ".xml");
+      if (!settings.m_overwrite && CFile::Exists(xmlFile))
+        xmlFile = URIUtils::AddFileToFolder(strFolder, "kodi_musicdb" + CDateTime::GetCurrentDateTime().GetAsSaveString() + ".xml");
+      xmlDoc.SaveFile(xmlFile);
+
+      CVariant data;
+      if (settings.GetSinglefile())
+      {
+        data["file"] = xmlFile;
+        if (iFailCount > 0)
+          data["failcount"] = iFailCount;
+      }
+      ANNOUNCEMENT::CAnnouncementManager::GetInstance().Announce(ANNOUNCEMENT::AudioLibrary, "xbmc", "OnExport", data);
     }
-    ANNOUNCEMENT::CAnnouncementManager::GetInstance().Announce(ANNOUNCEMENT::AudioLibrary, "xbmc", "OnExport", data);
   }
   catch (...)
   {
     CLog::Log(LOGERROR, "%s failed", __FUNCTION__);
     iFailCount++;
   }
 
-  if (progress)
-    progress->Close();
+  if (progressDialog != NULL)
```

Comment:
`nullptr` is preferred over `NULL`, but `if (progressDialog)` should be enough here and everywhere.